### PR TITLE
refactor: re-define module structs for serialization

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
The details in https://github.com/aquasecurity/trivy/issues/6654

## Related issues
- Close https://github.com/aquasecurity/trivy/issues/6654

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/module/serialize/types.go [hunk 1]
```diff
@@ -39,4 +39,98 @@ type PostScanSpec struct {
...
+	VulnerabilityID  string         `json:",omitempty"`
+	VendorIDs        []string       `json:",omitempty"`
+	PkgID            string         `json:",omitempty"`
+	PkgName          string         `json:",omitempty"`
+	PkgPath          string         `json:",omitempty"`
+	InstalledVersion string         `json:",omitempty"`
+	FixedVersion     string         `json:",omitempty"`
+	Status           types.Status   `json:",omitempty"`
+	Layer            Layer          `json:",omitempty"`
+	SeveritySource   types.SourceID `json:",omitempty"`
```
Comment (DmitriyLewen) [inline pkg/module/serialize/types.go:62]: Perhaps we want re-defines Status and SourceID to avoid importing of trivy-db.
wdyt?

Comment (knqyf263) [reply pkg/module/serialize/types.go:62]: Trivy DB doesn't have so many dependencies at the moment. It's better to reuse structs as much as possible.

Comment (DmitriyLewen) [reply pkg/module/serialize/types.go:62]: Got it.

**Review by DmitriyLewen — APPROVED** [review]

File: pkg/module/serialize/types.go (hunk 1 above)
Comment (knqyf263) [reply pkg/module/serialize/types.go:62]: Once we struggle with the dependencies in trivy-db, yes, I'll remove the dependency of trivy-db.

**Pull request merged**